security/src/main/java/io/airlift/security/client/UrlUtil.java

```diff
@@ -0,0 +1,22 @@
+package io.airlift.security.client;
+
+import com.google.common.base.Throwables;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * url utility functions shared within the package.
+ */
+class UrlUtil
+{
+    public static URI normalizedUri(URI uri)
+    {
+        try {
+            return new URI(uri.getScheme(), null, uri.getHost(), uri.getPort(), null, null, null);
+        }
+        catch (URISyntaxException e) {
+            throw Throwables.propagate(e);
```

Comment:
No, this is more like an AssertionError. It should be impossible to build a broken URI from a valid URI in the manner done in this method.